packages/gatsby-plugin-utils/src/polyfill-remote-file/index.ts

```diff
@@ -16,15 +16,16 @@ import {
   gatsbyImageResolver,
 } from "./graphql/gatsby-image-resolver"
 
-import type { Actions } from "gatsby"
+import type { Actions, Store } from "gatsby"
 import type { InterfaceTypeComposerAsObjectDefinition } from "graphql-compose"
 import type { SchemaBuilder, IRemoteFileNode } from "./types"
 
 let enums: ReturnType<typeof getRemoteFileEnums> | undefined
 
 export function getRemoteFileFields(
   enums: ReturnType<typeof getRemoteFileEnums>,
-  actions: Actions
+  actions: Actions,
+  store: Store
```

Comment:
```suggestion
  store?: Store
```

Adding requirement that `store` should be passed is a breaking change. We should have defensive code that check if `store` is there and bailout of anything extra that is happening in this PR if it's not (we can warn but we shouldn't fail, especially if user doesn't actually need those extra headers). I did see some defensive code across this PR - but we can utilize typescript (when it's used) to help us out with pointing those places to us.

Change like that likely need

Reply:
Good call! I'll make that change and see what TS tells me :)

Reply:
Looks like the defensive code that was in there was good enough, but now TS tells us for sure! 🙌